rpmlint/checks/BinariesCheck.py

```diff
@@ -25,6 +25,48 @@ def create_nonlibc_regexp_call(call):
     r = r'.*?\s+UND\s+(%s)\s?.*$' % call
     return re.compile(r)
 
+class Issue:
+    """
+    A simple issues wrapper that later reported via add_info.
+    """
+
+    def __init__(self, level, reason, description = None):
+        self.level = level
+        self.reason = reason
+        self.description = description
+
+class BinaryInfoChecker:
+    def __init__(self, path):
+        self.path = path
+        self.is_archive = path.endswith('.a')
+        self.issues = []
+        self.readelf_parser = ReadelfParser(path)
+        # register all check functions
+        self.check_functions = [self.check_lto_section, self.check_no_text_in_archive]
+
+    # For an archive, test if any .text sections is empty
+    def check_no_text_in_archive(self):
+        if self.is_archive:
+            for elf_file in self.readelf_parser.section_info.elf_files:
+                for section in elf_file:
+                    if section.name == '.text' and section.size == 0:
+                        self.issues.append(Issue('E', 'lto-no-text-in-archive', self.path))
```

Comment:
Why not right-away call the filter add_info?

Reply:
Again here, it's about the possibility to test it easily.

Reply:
We discussed this in the office to use the output.results content :)